src/test/java/com/pusher/client/channel/impl/PresenceChannelImplTest.java

```diff
@@ -107,6 +107,27 @@ public void testInternalSubscriptionSucceededMessageIsTranslatedToASubscriptionS
     }
 
     @Test
+    public void testMemberAddedMessageLeadsToUserIdBeingStoredCorrectly() {
+        final Map<String, String> userInfo = new LinkedHashMap<String, String>();
+        userInfo.put("name", "Phil Leggetter");
+        userInfo.put("twitter_id", "@leggetter");
+
+        final Map<String, Object> data = new LinkedHashMap<String, Object>();
+        data.put("user_id", 123);
+        data.put("user_info", userInfo);
+
+        final String eventName = "pusher_internal:member_added";
+
+        channel.onMessage(eventName, eventJson(eventName, data, getChannelName()));
```

Comment:
:+1: Passing in the JSON here seems the right thing to do. It seems consistent with the other tests and then exercises the bit of code which is causing problems.